### DISCUSSION TITLE
Is `textDocument/formatting` support planned?

### DISCUSSION THREAD

**Discussion by vendion** [phpactor/phpactor#1851] [Q&A] [answered]
Is adding in a handler for the `textDocument/formatting` method to phpactor planned, or is this seen as being out of scope for phpactor?

**Comment by dantleech** [comment]
Notnplanned now.If we did it would be a proxy for php-cs-fixrr or phpcs - although not sure how we can make those tools work on a selection

**Comment by vendion** [comment]
Having it be a proxy for `php-cs-fixer` could work, as IIRC it defaults to PSR12 standard with the ability to set a different rule set. I'm not too familiar with the LSP spec, but isn't format whole document and selection handled differently (I could be wrong as I haven't read the whole spec).

**Reply by dantleech** [reply]
Maybe, worth looking into. This would be similar to the phpstan and psalm extensions for diagnostics.

**Reply by dantleech** [reply]
I made a prototype, the spec does indeed support both full document and selection formatting. As with the `phpstan` and `pslam` extensions it would depend on your project using the tool (i.e.  `vendor/bin/php-cs-fixer`) although it would be possible to specify the path if you had to use a global installation for some reason.

**Comment by dantleech** [comment, marked as answer]
There is now an extension to support PHP-CS-Fixer for now: https://phpactor.readthedocs.io/en/master/integrations/php-cs-fixer.html
